File: app/src/main/java/com/example/jvmori/myweatherapp/di/modules/main/ViewModelFactoryModule.java
package com.example.jvmori.myweatherapp.di.modules.main;
import androidx.lifecycle.ViewModelProvider;
import com.example.jvmori.myweatherapp.di.scope.MainActivityScope;
import com.example.jvmori.myweatherapp.ui.viewModel.ViewModelProviderFactory;


import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelFactoryModule {
    @Binds
    @MainActivityScope
    abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelProviderFactory modelProviderFactory);
}

File: app/src/main/java/com/example/jvmori/myweatherapp/di_/modules/ViewModelFactoryModule.kt
package com.example.jvmori.myweatherapp.di_.modules

import androidx.lifecycle.ViewModelProvider
import com.example.jvmori.myweatherapp.ui.viewModel.ViewModelProviderFactory
import dagger.Binds
import dagger.Module

@Module
abstract class ViewModelFactoryModule {
    @Binds
    abstract fun bindViewModelFactory(modelProviderFactory: ViewModelProviderFactory) : ViewModelProvider.Factory
}